src/pages/profile/index.js

```diff
@@ -0,0 +1,205 @@
+import React, { useState, useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { navigate, Link } from 'gatsby'
+import { useFormik } from 'formik'
+import * as Yup from 'yup'
+import Layout from '../../components/Layout'
+import { getUser } from '../../services/util'
+
+import { getProfile, addProfile } from '../../services/api'
+
+const MyProfile = () => {
+  const { t } = useTranslation()
+  const user = getUser()
+  const [error, setError] = useState(null)
+  const [profile, setProfile] = useState({})
+  const [isSaved, setIsSaved] = useState(false)
+
+  useEffect(() => {
+    if (!user) {
+      navigate('/signin')
+    }
+    const getProfileData = async () => {
+      try {
+        const { data } = await getProfile()
+        if (data) {
+          setProfile(data)
+        }
+        setError(null)
+      } catch (err) {
+        if (err.message.match(/(403|400)/)) {
+          setError('You are not a user')
+        } else setError('Something went wrong')
```

Comment:
1- Can we change those errors to be coming from the messages files, please
2- Instead of using the state to keep track of /display the errors, can you please use the `react-toastify` library instead. you can remove the `error, isSaved` state members and use **toast.success** and **toast.error** instead to display a notification

Usage Reference: https://github.com/NyalaDev/edu-ui/blob/master/src/pages/signup/enhancedForm.js#L26

Reply:
I was using `error` to display a link to the signIn page and hide the form, how to do that with `react-toast`?

Reply:
I think we should not display/show this page if the user is not signed in, right!

Reply:
Yes, It will not show, but I though in case they already have the link to it.
Doesn't matter the user will get `you are not a user` error after submitting the form anyway